spec.html

```diff
@@ -9599,17 +9614,11 @@ <h1>
         1. Let _n_ be 0.
         1. Repeat,
           1. If _iteratorRecord_.[[Done]] is *false*, then
-            1. Let _next_ be Completion(IteratorStep(_iteratorRecord_)).
-            1. If _next_ is an abrupt completion, set _iteratorRecord_.[[Done]] to *true*.
-            1. ReturnIfAbrupt(_next_).
-            1. If _next_ is *false*, set _iteratorRecord_.[[Done]] to *true*.
+            1. Let _next_ be ? IteratorStepValue(_iteratorRecord_).
           1. If _iteratorRecord_.[[Done]] is *true*, then
             1. If _environment_ is *undefined*, return ? PutValue(_lhs_, _A_).
             1. Return ? InitializeReferencedBinding(_lhs_, _A_).
-          1. Let _nextValue_ be Completion(IteratorValue(_next_)).
-          1. If _nextValue_ is an abrupt completion, set _iteratorRecord_.[[Done]] to *true*.
-          1. ReturnIfAbrupt(_nextValue_).
-          1. Perform ! CreateDataPropertyOrThrow(_A_, ! ToString(𝔽(_n_)), _nextValue_).
```

Comment:
Maybe a `NOTE: _next_ is not ~done~`? The indented `Let _next_` inside the branch took me a second to grok.

Reply:
I’d expect an assert instead of a note, i think

Reply:
Done, but then also in https://github.com/tc39/ecma262/pull/3268/commits/1c73742b2f6b78b55670c976fbe975ea996cc4ac I swapped that for a potentially clearer implementation.

Reply:
New versions lgtm.